File: exercise/ts/day23/src/external/stable/magicStable.ts
import {Reindeer} from "../deer/reindeer";

export class MagicStable {
    // Dasher is sick since yesterday
    private dasher = new Reindeer("Dasher", 4, 10, true);
    private dancer = new Reindeer("Dancer", 2, 8);
    private prancer = new Reindeer("Prancer", 3, 9);
    private vixen = new Reindeer("Vixen", 3, 6);
    // Comet is sick as well
    private comet = new Reindeer("Comet", 4, 9, true);
    private cupid = new Reindeer("Cupid", 4, 6);
    private donner = new Reindeer("Donner", 7, 6);
    private blitzen = new Reindeer("Blitzen", 8, 7);
    private rudolph = new Reindeer("Rudolph", 6, 3);

    public getAllReindeers(): Reindeer[] {
        return [
            this.dasher, this.dancer, this.prancer, this.vixen,
            this.comet, this.cupid, this.donner, this.blitzen, this.rudolph
        ];
    }
}

File: exercise/scala/day23/src/main/scala/external/stable/MagicStable.scala
package external.stable

import external.deer.Reindeer

class MagicStable {
  // Dasher is sick since yesterday
  private val dasher = new Reindeer("Dasher", 4, 10, true)
  private val dancer = new Reindeer("Dancer", 2, 8)
  private val prancer = new Reindeer("Prancer", 3, 9)
  private val vixen = new Reindeer("Vixen", 3, 6)
  // Comet is sick as well
  private val comet = new Reindeer("Comet", 4, 9, true)
  private val cupid = new Reindeer("Cupid", 4, 6)
  private val donner = new Reindeer("Donner", 7, 6)
  private val blitzen = new Reindeer("Blitzen", 8, 7)
  private val rudolph = new Reindeer("Rudolph", 6, 3)

  def getAllReindeers: List[Reindeer] = {
    List(dasher, dancer, prancer, vixen, comet, cupid, donner, blitzen, rudolph)
  }
}